client/components/tinymce/plugins/media/drop-zone.jsx

```diff
@@ -91,12 +101,39 @@ export default React.createClass( {
 		} );
 	},
 
+	beforeInsertMedia( files ) {
+		if ( ! config.isEnabled( 'post-editor/image-reorientation' ) ) {
+			return Promise.resolve();
+		}
+
+		const isSingleImage = 1 === files.length && 'image' === MediaUtils.getMimePrefix( files[ 0 ] );
+
+		// This is a little tricky. If there's a single image that requires
+		// re-orientation (due to exif data), then we need to block *all* saving
+		// AND the updating of transient images within TinyMCE's editor.
+		// This is because when an image is dragged in the drop-zone, if it needs
+		// re-orientation then it is NOT inserted into the DOM until re-orientation
+		// finishes.
+		// Once re-orientation completes, we can unblock saving (handled in `afterInsertMedia`)
+		if ( isSingleImage ) {
+			return getImageOrientation( files[0] )
```

Comment:
Technically there's an exception for spacing of literal arguments to arrays ([reference](https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/javascript.md#arrays-and-function-calls)) which is occasionally ignored, but we should at least be consistent within the same file (see a few lines above where you'd added spaces to `files[ 0 ]`)